src/linker/Linker/MethodReferenceExtensions.cs

```diff
@@ -10,38 +10,36 @@ public static class MethodReferenceExtensions
 	{
 		public static string GetDisplayName (this MethodReference method)
 		{
-			var builder = new System.Text.StringBuilder ();
-			builder.Append ((method as MemberReference).GetNamespaceDisplayName ());
-			builder.Append ('.');
-			builder.Append (method.DeclaringType.GetDisplayName ());
-			builder.Append ("::");
-			if (method.Name == ".ctor")
-				builder.Append (method.DeclaringType.Name);
-			else
-				builder.Append (method.Name);
-
-			if (method.HasGenericParameters) {
-				builder.Append ('<');
+			var sb = new System.Text.StringBuilder ();
 
-				for (int i = 0; i < method.GenericParameters.Count - 1; i++)
-					builder.Append ($"{method.GenericParameters[i].GetDisplayName ()},");
+			// Append parameters
+			sb.Append ("(");
+			if (method.HasParameters) {
+				for (int i = 0; i < method.Parameters.Count - 1; i++)
+					sb.Append (method.Parameters[i].ParameterType.GetDisplayName ()).Append (',');
 
-				builder.Append ($"{method.GenericParameters[method.GenericParameters.Count - 1].GetDisplayName ()}>");
+				sb.Append (method.Parameters[method.Parameters.Count - 1].ParameterType.GetDisplayName ());
 			}
 
-			builder.Append ("(");
-
-			if (method.HasParameters) {
-				for (int i = 0; i < method.Parameters.Count - 1; i++) {
-					builder.Append ($"{method.Parameters[i].ParameterType.GetDisplayName ()},");
-				}
-
-				builder.Append (method.Parameters[method.Parameters.Count - 1].ParameterType.GetDisplayName ());
+			sb.Append (")");
+			
+			// Insert generic parameters
+			if (method.HasGenericParameters) {
+				TypeReferenceExtensions.ParseGenericParameters (method.GenericParameters, null, method.GenericParameters.Count, sb);
 			}
 
-			builder.Append (")");
+			// Insert method name
```

Comment:
Why do you prepend the name at the end, why not simply append it before the parameters?